airflow/models/taskinstance.py

```diff
@@ -1009,19 +1029,27 @@ def signal_handler(signum, frame):
                     self.log.error("Failed when executing execute callback")
                     self.log.exception(e3)
 
-                # If a timeout is specified for the task, make it fail
-                # if it goes beyond
-                result = None
-                if task_copy.execution_timeout:
-                    try:
-                        with timeout(int(
-                                task_copy.execution_timeout.total_seconds())):
+                with ExitStack() as exit_stack:
+                    from airflow.task.context.current import set_current_context
+                    exit_stack.enter_context(set_current_context(context))
+                    user_defined_exec_context = get_user_defined_execute_context()
```

Comment:
I would give access to context and self in get_user_defined_execute_context() implementation, so if user code needs context - it can get it